### DISCUSSION TITLE
Cmake error when applying mujoco dependency libraries

### DISCUSSION THREAD

**Discussion by yzd0014** [google-deepmind/mujoco#469] [Asking for Help]
I got the following cmake error that only occurs on one of my computers. After a lot searching, it seems like the error happens when doing "git apply" and there is unknowing command line option injected which is "-q". The library that cmake is trying do get also starts with letter "q", I'm not sure if this is the cause. Overall I still didn't connects dots together, any help would be really appreciated!

Found Git: C:/Program Files/Git/cmd/git.exe (found version "2.29.2.windows.2") 
mujoco::FindOrFetch: checking for targets in package `qhull`
mujoco::FindOrFetch: checking for targets in package `qhull` - target `qhull` not defined.
mujoco::FindOrFetch: Using FetchContent to retrieve `qhull`
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Creating directories for 'qhull-populate'
  Building Custom Rule C:/Users/10517/Desktop/mujoco-main/build/_deps/qhull-subbuild/CMakeLists.txt
  Performing download step (git clone) for 'qhull-populate'
  Cloning into 'qhull-src'...
  HEAD is now at 3df027b Merge pull request #103 from cmF3/master
  Performing update step for 'qhull-populate'
  Performing patch step for 'qhull-populate'
CUSTOMBUILD : error : unknown switch `q' [C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\qhull-populate.vcxproj]
  usage: git apply [<options>] [<patch>...]
  
      --exclude <path>      don't apply changes matching the given path
      --include <path>      apply changes matching the given path
      -p <num>              remove <num> leading slashes from traditional diff paths
      --no-add              ignore additions made by the patch
      --stat                instead of applying the patch, output diffstat for the input
      --numstat             show number of added and deleted lines in decimal notation
      --summary             instead of applying the patch, output a summary for the input
      --check               instead of applying the patch, see if the patch is applicable
      --index               make sure the patch is applicable to the current index
      -N, --intent-to-add   mark new files with `git add --intent-to-add`
      --cached              apply a patch without touching the working tree
      --unsafe-paths        accept a patch that touches outside the working area
      --apply               also apply the patch (use with --stat/--summary/--check)
      -3, --3way            attempt three-way merge if a patch does not apply
      --build-fake-ancestor <file>
                            build a temporary index based on embedded index information
      -z                    paths are separated with NUL character
      -C <n>                ensure at least <n> lines of context match
      --whitespace <action>
                            detect new or modified lines that have whitespace errors
      --ignore-space-change
                            ignore changes in whitespace when finding context
      --ignore-whitespace   ignore changes in whitespace when finding context
      -R, --reverse         apply the patch in reverse
      --unidiff-zero        don't expect at least one line of context
      --reject              leave the rejected hunks in corresponding *.rej files
      --allow-overlap       allow overlapping hunks
      -v, --verbose         be verbose
      --inaccurate-eof      tolerate incorrectly detected missing new-line at the end of file
      --recount             do not trust the line counts in the hunk headers
      --directory <root>    prepend <root> to all filenames
  
  No configure step for 'qhull-populate'
  No build step for 'qhull-populate'
  No install step for 'qhull-populate'
  No test step for 'qhull-populate'
  Completed 'qhull-populate'
C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Microsoft\VC\v170\Microsoft.CppCommon.targets(242,5): error MSB8066: Custom build for 'C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\a67b01c64e04f6f46133660fae4c5c7b\qhull-populate-mkdir.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\a67b01c64e04f6f46133660fae4c5c7b\qhull-populate-download.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\a67b01c64e04f6f46133660fae4c5c7b\qhull-populate-update.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\a67b01c64e04f6f46133660fae4c5c7b\qhull-populate-patch.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\a67b01c64e04f6f46133660fae4c5c7b\qhull-populate-configure.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\a67b01c64e04f6f46133660fae4c5c7b\qhull-populate-build.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\a67b01c64e04f6f46133660fae4c5c7b\qhull-populate-install.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\a67b01c64e04f6f46133660fae4c5c7b\qhull-populate-test.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\dd5033c9d2ec3946dcf793a0dc309446\qhull-populate-complete.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeFiles\83bd356241111478baf83fc2c2feefd0\qhull-populate.rule;C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\CMakeLists.txt' exited with code -1. [C:\Users\10517\Desktop\mujoco-main\build\_deps\qhull-subbuild\qhull-populate.vcxproj]

CMake Error at C:/Program Files/CMake/share/cmake-3.23/Modules/FetchContent.cmake:1088 (message):
  Build step for qhull failed: 1
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.23/Modules/FetchContent.cmake:1217:EVAL:2 (__FetchContent_directPopulate)
  C:/Program Files/CMake/share/cmake-3.23/Modules/FetchContent.cmake:1217 (cmake_language)
  cmake/FindOrFetch.cmake:125 (FetchContent_Populate)
  cmake/MujocoDependencies.cmake:107 (findorfetch)
  CMakeLists.txt:53 (include)

Configuring incomplete, errors occurred!
See also "C:/Users/10517/Desktop/mujoco-main/build/CMakeFiles/CMakeOutput.log".

**Comment by saran-t** [comment]
Looks like a duplicate of https://github.com/deepmind/mujoco/issues/448 -- please continue to post over there instead.

Also, for questions and bug reports like this, please use Issues rather than Discussions. We are planning to deprecate this Q&A section.
